include/Beman/Optional26/optional.hpp

```diff
@@ -309,6 +327,11 @@ class optional {
     }
 
   public:
+    using value_type = T;
+    // Since P3168R1: Give std::optional Range Support.
+    using iterator       = T*;       // see [optional.iterators]
+    using const_iterator = const T*; // see [optional.iterators]
```

Comment:
It would be better to not use pointers.

Reply:
@brevzin , I agree with you and I have a WIP patch for an iterator implementation. I would like to put this work in a new PR, if that's OK with you.